### ISSUE TITLE
subclassing Conv?

### ISSUE THREAD

**Issue by APJansen** [keras-team/keras#15352]
I am working on an implementation of group convolutions in Keras, after [Cohen & Welling 2016](https://arxiv.org/abs/1602.07576). For 2D group-convolutions I've subclassed `Conv2D` as `GroupConv2D`, now I want to add 3D group-convolutions, and it makes more sense to subclass `Conv` as `GroupConv`, and then subclass that as `GroupConv2D` and `GroupConv3D`, as is done with regular convolutions in Keras. However `Conv` itself is not exported so I can't do that.

Is it possible to make `Conv` public? Or is there perhaps a different way for me to do this without `Conv` being public? Without using the tensorflow backend I can import it (`from keras.layers.convolutional import Conv` works but `from tensorflow.keras.layers.convolutional import Conv` does not), but that seems like a bad idea, or is that fine?

**Comment by innat** [comment]
FYI, I think [group convolution](https://keras.io/api/layers/convolution_layers/convolution3d/) is supported now. See the documents. 

```python
tf.keras.layers.Conv3D(
    filters,
    kernel_size,
    strides=(1, 1, 1),
    padding="valid",
    data_format=None,
    dilation_rate=(1, 1, 1),
    groups=1, # < ----------
    activation=None,
    use_bias=True,
    kernel_initializer="glorot_uniform",
    bias_initializer="zeros",
    kernel_regularizer=None,
    bias_regularizer=None,
    activity_regularizer=None,
    kernel_constraint=None,
    bias_constraint=None,
    **kwargs
)
```

> **groups**: A positive integer specifying the number of groups in which the input is split along the channel axis. Each group is convolved separately with filters / groups filters. The output is the concatenation of all the groups results along the channel axis. Input channels and filters must both be divisible by groups.

**Comment by APJansen** [comment]
Sorry I should have clarified this because it's a confusing naming conflict. The groups argument you point to I think refers to what I've seen called group_ed_ convolutions. I'm not familiar with the details  of this, but a group is something like a subset of the channels. While in the group convolution I'm referring to (see link above), the group refers to a symmetry group. Group convolutions (rather than grouped convolutions) are equivariant to more symmetry transformations than just translations, for instance 90 degree rotations.

**Comment by LukeWood** [comment]
Hey @APJansen - how much logic would be able to be shared between GroupConv2D and GroupConv3D, perhaps we can expose it but typically we require a pretty strong case to expose base classes.

+fchollet

**Comment by APJansen** [comment]
Hi @LukeWood, like the standard Conv2D and Conv3D layers: literally everything except for a rank attribute being set to 2 or 3 respectively. 
I've worked around it though by moving all functionality that doesn't require a call to super to a separate layer, limiting the code duplication. (see https://github.com/apjansen/groco if you're curious)

So if there are good reasons not to expose the base Conv class, no real need to do it on my account.
Thanks!

**Comment by AdityaKane2001** [comment]
@APJansen 

I'm not sure, but will [this](https://chromium.googlesource.com/external/github.com/tensorflow/tensorflow/+/HEAD/RELEASE.md#release-2_7_0-major-features-and-improvements) help? (See "Keras" under "Major Features and Improvements")

**Comment by APJansen** [comment]
@AdityaKane2001 The way I've done it now (see above) is still cleaner I think, but it won't work for separable/depthwise convolutions, so this may be useful for those, thanks for letting me know!

**Comment by fchollet** [comment]
Overriding the `convolution_op` method of `tf.keras.layers.Conv2D`/etc is what I would recommend doing at this time. We don't intend to make the `Conv` base layer public, though you could always maintain your own copy of it as part of your codebase.
